### ISSUE TITLE
Android: [FATAL ERROR: Could not find system resources]

### ISSUE THREAD

**Issue by erlanger** [SWI-Prolog/swipl-devel#507]
Some people are having a strange problem in some android devices, 
see this issue: https://github.com/erlanger/swipl-termux/issues/5

The error is  [FATAL ERROR: Could not find system resources]
but `boot.prc`  file is found as the strace shows:

```
openat(AT_FDCWD, "/data/data/com.termux/files/usr/lib/swipl/boot.prc", O_RDONLY|O_LARGEFILE) = 3
fstat64(3, {st_mode=S_IFREG|0600, st_size=85604, ...}) = 0
close(3) = 0
gettimeofday({tv_sec=1568916005, tv_usec=857728}, NULL) = 0
openat(AT_FDCWD, "/dev/properties/u:object_r:system_prop:s0", O_RDONLY|O_LARGEFILE|O_NOFOLLOW|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0444, st_size=131072, ...}) = 0
mmap2(NULL, 131072, PROT_READ, MAP_SHARED, 3, 0) = 0xb0937000
close(3) = 0
openat(AT_FDCWD, "/data//misc/zoneinfo/current/tzdata", O_RDONLY|O_LARGEFILE) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/system//usr/share/zoneinfo/tzdata", O_RDONLY|O_LARGEFILE) = 3
read(3, "tzdata2016g\0\0\0\0\30\0\0w\274\0\7I\270", 24) = 24
lseek(3, 24, SEEK_SET) = 24
read(3, "Africa/Abidjan\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 30628) = 30628
lseek(3, 395734, SEEK_SET) = 395734
read(3, "TZif2\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\7\0\0\0\7\0\0\0\0"..., 3661) = 3661
close(3) = 0
openat(AT_FDCWD, "/data//misc/zoneinfo/current/tzdata", O_RDONLY|O_LARGEFILE) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/system//usr/share/zoneinfo/tzdata", O_RDONLY|O_LARGEFILE) = 3
read(3, "tzdata2016g\0\0\0\0\30\0\0w\274\0\7I\270", 24) = 24
lseek(3, 24, SEEK_SET) = 24
read(3, "Africa/Abidjan\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0"..., 30628) = 30628
close(3) = 0
write(2, "[FATAL ERROR: at Thu Sep 19 19:0"..., 43) = 43
write(2, "Could not find system resources", 31) = 31
write(2, "]\n", 2) = 2
rt_sigprocmask(SIG_SETMASK, [HUP INT QUIT ILL TRAP BUS FPE KILL USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT STOP TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH IO PWR SYS RTMIN], [], 8) = 0
tgkill(16559, 16559, SIGABRT) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=16559, si_uid=10151} ---
sigaction(SIGABRT, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0xb0c48b98}, {sa_handler=0xb0c03825, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_RESTART|SA_SIGINFO, sa_restorer=0xb0c48ba4}) = 0
sigaction(SIGABRT, {sa_handler=0xb0c03825, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_RESTART|SA_SIGINFO, sa_restorer=0xb0c48ba4}, NULL) = 0
prctl(PR_GET_NAME, "swipl") = 0
openat(AT_FDCWD, "/dev/properties/u:object_r:log_tag_prop:s0", O_RDONLY|O_LARGEFILE|O_NOFOLLOW|O_CLOEXEC) = 3
fstat64(3, {st_mode=S_IFREG|0444, st_size=131072, ...}) = 0
mmap2(NULL, 131072, PROT_READ, MAP_SHARED, 3, 0) = 0xb0917000
close(3) = 0
socket(AF_UNIX, SOCK_DGRAM|SOCK_CLOEXEC, 0) = 3
fcntl64(3, F_SETFL, O_RDONLY|O_NONBLOCK) = 0
connect(3, {sa_family=AF_UNIX, sun_path="/dev/socket/logdw"}, 110) = 0
openat(AT_FDCWD, "/dev/properties/u:object_r:default_prop:s0", O_RDONLY|O_LARGEFILE|O_NOFOLLOW|O_CLOEXEC) = 4
fstat64(4, {st_mode=S_IFREG|0444, st_size=131072, ...}) = 0
mmap2(NULL, 131072, PROT_READ, MAP_SHARED, 4, 0) = 0xb0751000
close(4) = 0
openat(AT_FDCWD, "/dev/properties/u:object_r:logd_prop:s0", O_RDONLY|O_LARGEFILE|O_NOFOLLOW|O_CLOEXEC) = 4
fstat64(4, {st_mode=S_IFREG|0444, st_size=131072, ...}) = 0
mmap2(NULL, 131072, PROT_READ, MAP_SHARED, 4, 0) = 0xb0731000
close(4) = 0
clock_gettime(CLOCK_REALTIME, {tv_sec=1568916005, tv_nsec=863962169}) = 0
writev(3, [{iov_base="\4", iov_len=1}, {iov_base="\257@", iov_len=2}, {iov_base="%\302\203]9\4\1773", iov_len=8}, {iov_base="\7", iov_len=1}, {iov_base="libc\0", iov_len=5}, {iov_base="Fatal signal 6 (SIGABRT), code -"..., iov_len=55}], 6) = 72
close(3) = 0
socket(AF_UNIX, SOCK_STREAM|SOCK_CLOEXEC, 0) = 3
connect(3, {sa_family=AF_UNIX, sun_path=@"android:debuggerd"}, 20) = 0
write(3, "\0\0\0\0\257@\0\0\0\0\0\0\0\0\0\0\372\377\377\377", 20) = 20
read(3, "", 1) = 0
close(3) = 0
sigaction(SIGABRT, {sa_handler=SIG_DFL, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART, sa_restorer=0xb0c48b98}, {sa_handler=0xb0c03825, sa_mask=[], sa_flags=SA_RESTORER|SA_ONSTACK|SA_RESTART|SA_SIGINFO, sa_restorer=0xb0c48ba4}) = 0
rt_tgsigqueueinfo(16559, 16559, SIGABRT, {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=16559, si_uid=10151}) = 0
rt_sigreturn({mask=[HUP INT QUIT ILL TRAP BUS FPE USR1 SEGV USR2 PIPE ALRM TERM STKFLT CHLD CONT TSTP TTIN TTOU URG XCPU XFSZ VTALRM PROF WINCH IO PWR SYS RTMIN]}) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=16559, si_uid=10151} ---
+++ killed by SIGABRT +++
M
```

`boot.prc` is found, but it is not mmap'ed after the `fstat64(...)` call.

Do you have any idea what this could be?

**Comment by JanWielemaker** [comment]
The most sensible thing I can think of is that the file is corrupted and not recognised as a zip archive?

**Comment by erlanger** [comment]
I asked them to uninstall  swi-prolog and to reinstall it. Also asked them to uninstall termux completely and reinstall everything again; I doubt the file would get corrupted twice in a row....
This is a mystery...it only happens to very few people.

**Comment by erlanger** [comment]
Fixed by PR #508
